### ISSUE TITLE
Header cannot be written in producer message

### ISSUE THREAD

**Issue by TheZwieback** [tulios/kafkajs#546]
Hi,
The header and message types are defined as:

```
export interface Message {
  key?: Buffer | string | null
  value: Buffer | string | null
  partition?: number
  headers?: IHeaders
  timestamp?: string
}

export interface IHeaders {
  [key: string]: Buffer
}

```

This excludes writing headers as key value pairs as described in the documentation, see https://kafka.js.org/docs/producing#message-headers . Is there a way to fix this currently?

Thanks,
Christian

**Comment by tulios** [comment]
Hi @TheZwieback we should accept strings as well, can you create a PR and fix the `IHeaders` type? The Typescript definitions arrived late in the project and are still not quite right.

optionally you can `Buffer.from('string')` the values for now.

**Comment by TheZwieback** [comment]
Thanks for the quick reply. I changed the type definition and created a pull request.

**Comment by Nevon** [comment]
Fixed by #547
